dlt/normalize/items_normalizers.py

```diff
@@ -52,6 +63,217 @@ def __init__(
+
+        # Build new selects list in correct schema column order
+        new_selects = []
+        for col in columns:
+            lower_col = col.lower()
+            expr = alias_map.get(lower_col)
+            if expr:
+                new_selects.append(expr)
+            # If there's no such column select, just put null
+            else:
+                new_selects.append(
+                    sqlglot.exp.Alias(
+                        this=sqlglot.exp.Null(), alias=sqlglot.exp.to_identifier(casefolder_f(col))
+                    )
+                )
+
+        outer_parsed_select.set("expressions", new_selects)
+
+    def _build_outer_select_statement(
+        self,
+        parsed_select: sqlglot.exp.Select,
+    ) -> sqlglot.exp.Select:
+        """
+        Wraps the parsed SELECT statement in a subquery and returns the outer SELECT statement.
+        """
+        # Wrap parsed select in a subquery
+        subquery = parsed_select.subquery(alias="subquery")
+
+        # Get normalizer function
+        norm_f = self.schema.naming.normalize_identifier
+
+        # Get casefolder
+        casefold_f = self.config.destination_capabilities.casefold_identifier
+
+        # Build new select list using selected columns in the subquery
+        outer_selects: List[sqlglot.exp.Expression] = []
+        for select in parsed_select.selects:
+            if isinstance(select, sqlglot.exp.Star):
+                # NOTE: we could also replace star schema with explicit columns from schema
+                outer_selects.append(sqlglot.exp.Star(this="subquery"))
+                break
+            elif isinstance(select, sqlglot.exp.Alias):
+                name = select.alias
+                norm_casefolded = casefold_f(norm_f(name))
+                outer_selects.append(sqlglot.column(name, table="subquery").as_(norm_casefolded))
+                # NOTE: for bigquery, quoted identifiers are treated as Dot
+            elif isinstance(select, sqlglot.exp.Column) or isinstance(select, sqlglot.exp.Dot):
+                name = select.output_name or select.name
+                norm_casefolded = casefold_f(norm_f(name))
+                outer_selects.append(sqlglot.column(name, table="subquery").as_(norm_casefolded))
+
+        # Create the outer select statement
+        outer_select = sqlglot.select(*outer_selects).from_(subquery)
+
+        return outer_select
+
+    def __call__(self, extracted_items_file: str, root_table_name: str) -> List[TSchemaUpdate]:
+        with self.normalize_storage.extracted_packages.storage.open_file(
+            extracted_items_file, "r"
+        ) as f:
+            select_dialect, select_statement = read_dialect_and_sql(
+                file_obj=f,
+                fallback_dialect=self.config.destination_capabilities.sqlglot_dialect,  # caps are available at this point
+            )
+
+        parsed_select = sqlglot.parse_one(select_statement, read=select_dialect)
```

Comment:
We should probably raise here if this was not a select statement. Then you can just remove all the other tests for star selects in the individual steps. I think supporting these is too complicated at this point.